src/pages/TimeTravel.js

```diff
@@ -35,9 +35,9 @@ const TimeTravel = () => {
     .toISOString()
     .split("T")[0];
   const [selectedDate, setSelectedDate] = React.useState(yesterday);
-
   React.useEffect(() => {
-    const dayCount = timeTravelDate >= 0 ? timeTravelDate : getDayCount();
+    const dayCount =
+      timeTravelDate >= 0 && timeTravelDate < getDayCount() ? timeTravelDate : getDayCount();
```

Comment:
@avsssai could you pls update else condition to `getDatCount()-1` as the max they could is till yesterday, not today.

Reply:
@avsssai Also, It would be nice, If we could handle that future dates directly `handleChangeFromDate` method

Reply:
Sure, yeah it makes sense to do that there actually. Will change it, so by changing it to `getDayCount()-1` in the else, if they select a future date, we want them to go to yesterday instead of today?

Reply:
got it